### PR TITLE
Date are invalid if they are more than 5 years apart.

### LINKED ISSUE #89: Validate that the span of the calendar is under 10 - 20 years.
Have to see at what point it starts to get too slow.

### PR DESCRIPTION
**Pull request by andrayantelo**
Also fixed the allCalendarsId thing which you just watched me do. Fixes #89

### CODE REVIEW

File: calendar_reboot.js
Comment (johntyree) [inline calendar_reboot.js]: Why is this commented out?

File: calendar_reboot.js
Comment (johntyree) [inline calendar_reboot.js]: You don't need an if statement here. Just return the appropriate value.

File: calendar_reboot.js
Comment (johntyree) [inline calendar_reboot.js]: remove console.log

**Review by johntyree — CHANGES_REQUESTED** [review]

File: calendar_reboot.js
Comment (andrayantelo) [reply calendar_reboot.js]: I was checking to see where in the code allCalendarIds was getting overwritten.

Comment (andrayantelo) [reply calendar_reboot.js]: I uncommented it.

File: calendar_reboot.js
Comment (andrayantelo) [reply calendar_reboot.js]: Fixed.

File: calendar_reboot.js
Comment (andrayantelo) [reply calendar_reboot.js]: Removed.

**Comment by johntyree** [comment]
This branch has #88 in it, so don't merge this until #88 is merged. That's why the diff is so big.

If you want to fix that, then after you merge #88 into master, merge master back in to this branch.

**Review by johntyree — APPROVED** [review]

File: calendar_reboot.js
Comment (johntyree) [inline calendar_reboot.js]: `var` needed

File: calendar_reboot.js
Comment (johntyree) [inline calendar_reboot.js:697]: Nothing to change here, but in general this pattern is known as "read-modify-write" and it only works because we *know* that we are the only ones that are modifying the value.

If there were other code that could be modifying this, then you could end up "racing" where both of you read, then one of you modifies and writes but it gets "forgotten" when the second one writes because the version that the second one read didn't include the changes from the first.

**Review by johntyree — CHANGES_REQUESTED** [review]

**Review by johntyree — APPROVED** [review]

**Pull request merged**
